src/deprecated.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+/**
+ * @deprecated
+ */
+class_alias('League\JsonReference\DecoderInterface', 'League\JsonReference\JsonDecoderInterface');
```

Comment:
Instead of `class_alias` I would rather keep the existing files and extend the new one like this:

```php
/**
 * @deprecated use the DecoderInterface directly.
 */
interface JsonDecoderInterface extends DecoderInterface
{
}
```

That prevents the user from having 3 function calls for each request even if they aren't using the library and it's less confusing for IDEs.

You will have to remove final from `DecodingException` which is fine.

Reply:
Ok!